src/prefect/events/schemas/events.py

```diff
@@ -90,22 +94,37 @@ def role(self) -> str:
         return self["prefect.resource.role"]
 
 
+def _validate_related_resources(value) -> List:
+    from prefect.settings import PREFECT_EVENTS_MAXIMUM_RELATED_RESOURCES
+
+    if len(value) > PREFECT_EVENTS_MAXIMUM_RELATED_RESOURCES.value():
+        raise ValueError(
+            "The maximum number of related resources "
+            f"is {PREFECT_EVENTS_MAXIMUM_RELATED_RESOURCES.value()}"
+        )
+    return value
+
+
 class Event(PrefectBaseModel):
     """The client-side view of an event that has happened to a Resource"""
 
     model_config = ConfigDict(extra="ignore")
 
     occurred: DateTime = Field(
-        default_factory=lambda: pendulum.now("UTC"),
+        default_factory=lambda: DateTime.now("UTC"),
         description="When the event happened from the sender's perspective",
     )
     event: str = Field(
         description="The name of the event that happened",
+        max_length=1024,
```

Comment:
In general, I prefer validation like this only to exist server side since it won't apply to other clients (e.g. `curl` and other REST API clients) if the server side limit is increased. However, this is likely to affect a tiny set of users, if any, so this isn't a blocking comment; I just needed to get that off my chest.

Reply:
agreed, for consistency I removed client-side max_length here